tools/build_file_generator/JavaClassAnalyzer.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2018 The StartupOS Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.startupos.tools.buildfilegenerator;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.google.startupos.tools.buildfilegenerator.Protos.Import;
+import com.google.startupos.tools.buildfilegenerator.Protos.JavaClass;
+
+public class JavaClassAnalyzer {
+  public static JavaClass getJavaClass(String fileContent) {
+    JavaClass.Builder result = JavaClass.newBuilder();
+
+    List<String> importLines = getImportLines(fileContent);
+    importLines.forEach(line -> result.addImport(getImport(line)));
+
+    result.setIsTestClass(isTestClass(fileContent));
+    result.setHasMainMethod(hasMainMethod(fileContent));
+
+    return result.build();
+  }
+
+  private static List<String> getImportLines(String fileContent) {
+    return Arrays.stream(fileContent.split(System.lineSeparator()))
+        .filter(
+            line ->
```

Comment:
Maybe you can trim the string, to handle the case it starts with spaces?

Reply:
Done.